src/engine.rs

```diff
@@ -223,6 +223,7 @@ pub fn run(params: FunctionRunParams) -> Result<FunctionRunResult> {
         memory_usage,
         instructions,
         logs.to_string(),
+        String::from_utf8(input).unwrap(),
```

Comment:
Shouldn't ship this with an unwrap here- should use the `?` syntax: https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

That paired with our use of [anyhow](https://docs.rs/anyhow/latest/anyhow/) will ensure the errors raised by this are included as part of the method's result